casbin/effect/default_effector.py

```diff
@@ -7,33 +7,53 @@ class DefaultEffector(Effector):
     def merge_effects(self, expr, effects, results):
         """merges all matching results collected by the enforcer into a single decision."""
 
-        effects = set(effects)
         result = False
+
+        explain = False
+
+        explain_index = -1
+
+        if type(effects) == list:
+            explain = True
+
         if expr == "some(where (p_eft == allow))":
             if self.ALLOW in effects:
+                if explain:
+                    explain_index = effects.index(self.ALLOW)
                 result = True
 
         elif expr == "!some(where (p_eft == deny))":
             result = True
 
             if self.DENY in effects:
+                if explain:
+                    explain_index = effects.index(self.DENY)
                 result = False
 
         elif expr == "some(where (p_eft == allow)) && !some(where (p_eft == deny))":
             if self.DENY in effects:
+                if explain:
+                    explain_index = effects.index(self.ALLOW)
```

Comment:
Shouldn't it be `effects.index(self.DENY)`?

Reply:
Yes, My fault